pkg/controllers/subnet/subnet_controller.go

```diff
@@ -242,5 +244,58 @@ func (r *SubnetReconciler) Start(mgr ctrl.Manager) error {
 	if err != nil {
 		return err
 	}
+	go r.GarbageCollector(make(chan bool), servicecommon.GCInterval)
 	return nil
 }
+
+func (r *SubnetReconciler) GarbageCollector(cancel chan bool, timeout time.Duration) {
+	ctx := context.Background()
+	log.Info("subnet garbage collector started")
+	for {
+		select {
+		case <-cancel:
+			return
+		case <-time.After(timeout):
+		}
+		nsxSubnetList := r.Service.ListSubnet()
```

Comment:
Do we need to list all nsx subnets or just their IDs?

Reply:
subnet, GC will check its tag later